src/components/column-sorting/column-sorting.component.ts

```diff
@@ -0,0 +1,57 @@
+import { ColumnSortingDirective } from './column-sorting.directive';
+import { Component, Input, Output, EventEmitter, OnChanges, SimpleChanges, Host } from '@angular/core';
+import { BehaviorSubject } from 'rxjs/BehaviorSubject';
+import { Subject } from 'rxjs/Subject';
+
+@Component({
+    selector: 'ux-column-sorting',
+    templateUrl: './column-sorting.component.html',
+    exportAs: 'ux-column-sorting'
+})
+export class ColumnSortingComponent {
+    
+    @Input() state: ColumnSortingState;
```

Comment:
We should allow state to be a two way binding, so we need to add an output for stateChange that will emit an event whenever the state of this sorter changes